git head does not compile on FreeBSD

# syslog-ng

## Version of syslog-ng

git head snapshot as of June 16, 16:30

## Platform

FreeBSD 11.0 amd64

# Issue

## Failure

Does not compile. Using a self generated dist tgz from git head (used successfully on RPM distros) replacing 3.9.1 release source in /usr/ports/distfiles.

```
libtool: compile:  cc -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -I./modules/affile -I./modules/affile -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -pthread -I/usr/local/include/eventlog -I/usr/local/include -I/usr/include -I./lib/ivykis/src/include -I./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -pipe -fstack-protector -fno-strict-aliasing -Wall -pthread -MT modules/affile/modules_affile_libaffile_la-file-reader.lo -MD -MP -MF modules/affile/.deps/modules_affile_libaffile_la-file-reader.Tpo -c modules/affile/file-reader.c  -fPIC -DPIC -o modules/affile/.libs/modules_affile_libaffile_la-file-reader.o
mv -f modules/affile/.deps/modules_affile_libaffile_la-file-reader.Tpo modules/affile/.deps/modules_affile_libaffile_la-file-reader.Plo
/bin/sh ./libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I.  -I./lib -I./modules -I./lib -I./modules -I./modules/affile -I./modules/affile -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -pthread  -I/usr/local/include/eventlog  -I/usr/local/include  -I/usr/include   -I./lib/ivykis/src/include -I./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64  -O2 -pipe  -fstack-protector -fno-strict-aliasing -Wall -pthread -MT modules/affile/modules_affile_libaffile_la-wildcard-source.lo -MD -MP -MF modules/affile/.deps/modules_affile_libaffile_la-wildcard-source.Tpo -c -o modules/affile/modules_affile_libaffile_la-wildcard-source.lo `test -f 'modules/affile/wildcard-source.c' || echo './'`modules/affile/wildcard-source.c
libtool: compile:  cc -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -I./modules/affile -I./modules/affile -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -pthread -I/usr/local/include/eventlog -I/usr/local/include -I/usr/include -I./lib/ivykis/src/include -I./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -pipe -fstack-protector -fno-strict-aliasing -Wall -pthread -MT modules/affile/modules_affile_libaffile_la-wildcard-source.lo -MD -MP -MF modules/affile/.deps/modules_affile_libaffile_la-wildcard-source.Tpo -c modules/affile/wildcard-source.c  -fPIC -DPIC -o modules/affile/.libs/modules_affile_libaffile_la-wildcard-source.o
modules/affile/wildcard-source.c:328:60: error: use of undeclared identifier
      'O_LARGEFILE'
  self->file_reader_options.file_open_options.open_flags = DEFAULT_SD_OP...
                                                           ^
modules/affile/wildcard-source.c:28:67: note: expanded from macro
      'DEFAULT_SD_OPEN_FLAGS'
#define DEFAULT_SD_OPEN_FLAGS (O_RDONLY | O_NOCTTY | O_NONBLOCK | O_LARGEFILE)
                                                                  ^
1 error generated.
gmake[4]: *** [Makefile:11221: modules/affile/modules_affile_libaffile_la-wildcard-source.lo] Error 1
gmake[3]: *** [Makefile:14975: all-recursive] Error 1
gmake[2]: *** [Makefile:6107: all] Error 2
gmake[2]: Leaving directory '/usr/ports/sysutils/syslog-ng-devel/work/syslog-ng-3.9.1'
*** Error code 1
```

## Steps to reproduce

Compile syslog-ng git head on FreeBSD 11.0

--- a/modules/affile/affile-common.h
+++ b/modules/affile/affile-common.h
@@ -24,6 +24,9 @@
 #ifndef AFFILE_COMMON_H_INCLUDED
 #define AFFILE_COMMON_H_INCLUDED
 
+/* portable largefile support for affile */
+#include "compat/lfs.h"
+
 #include "file-perms.h"
 #include <string.h>
 

--- a/modules/affile/affile-dest.c
+++ b/modules/affile/affile-dest.c
@@ -33,7 +33,6 @@
 #include "logproto-file-writer.h"
 #include "transport/transport-file.h"
 #include "transport/transport-pipe.h"
-#include "compat/lfs.h"
 #include "logwriter.h"
 
 #include <iv.h>

--- a/modules/affile/file-reader.c
+++ b/modules/affile/file-reader.c
@@ -37,7 +37,6 @@
 #include "logproto-linux-proc-kmsg-reader.h"
 #include "poll-fd-events.h"
 #include "poll-file-changes.h"
-#include "compat/lfs.h"
 
 #include <sys/types.h>
 #include <sys/stat.h>

--- a/modules/affile/tests/test_affile_open_file.c
+++ b/modules/affile/tests/test_affile_open_file.c
@@ -24,7 +24,6 @@
 #include "testutils.h"
 #include "affile/affile-common.h"
 #include "lib/messages.h"
-#include "compat/lfs.h"
 #include <fcntl.h>
 #include <stdlib.h>
 #include <stdarg.h>

--- a/modules/affile/wildcard-source.h
+++ b/modules/affile/wildcard-source.h
@@ -24,6 +24,7 @@
 
 #include "syslog-ng.h"
 #include "driver.h"
+#include "affile-common.h"
 #include "file-reader.h"
 #include "directory-monitor.h"
 #include "directory-monitor-factory.h"

